deploy/docs/opentelemetry_collector.md

```diff
@@ -43,8 +43,18 @@ the same way for OpenTelemetry Collector like for Fluentd.
 
 ### Metrics Configuration
 
-All OpenTelemetry Collector configuration for metrics is located in
-[`values.yaml`][values] as `metadata.metrics.config`.
+There are two ways of directly configuring OpenTelemetry Collector for metrics metadata.
+
+The `metadata.metrics.config.merge` key can be used to provide configuration that will be merged
+with the Helm Chart's default configuration. It should be noted that this field is not subject to
+normal backwards compatibility guarantees, the default configuration can change even in minor
+versions while preserving the same end-to-end behaviour. Use of this field is discouraged - ideally
+the necessary customizations should be able to be achieved without touching the otel configuration
+directly. Please open an issue if your use case requires the use of this field.
+
+The `metadata.metrics.config.override` key can be used to provide configuration that will be completely
+replace the default configuration. This is an advanced feature requiring a good understanding of both
```

Comment:
> This is an advanced feature requiring a good understanding of this chart's architecture and OpenTelemetry Collector configuration

I like this warning and I think it should be applied to both `merge` and `override` properties.

Also, shouldn't the phrase "this field is not subject to normal backwards compatibility guarantees" be applied to both `merge` and `override` properties as well?

Reply:
I guess that for `override`, this depends on what you put there. You don't care at all about the configuration the Chart itself creates, but you might care about things like environment variables defined for OT Pods, or somesuch.

Reply:
I moved the warning to the start of the section and added a disclaimer about `override`.